### PR TITLE
CIRC-1504 "closed-filled" requests have "Request expiration" notices …

### PR DESCRIPTION
**Pull request by humoyunepam**
[CIRC-1504](https://issues.folio.org/browse/CIRC-1504)
## Purpose

Request expiration shouldn't be sent when request status is "Closed - Filled"
  

## Approach
- Added reschedule request notice check when check-out is made
- Wrote test
## Learning
Learned how scheduler and notices work.

### CODE REVIEW

File: src/test/java/api/requests/RequestScheduledNoticesProcessingTests.java [hunk 1]
```diff
@@ -155,6 +156,41 @@ void uponAtRequestExpirationNoticeShouldNotBeSentWhenRequestExpirationDateHasPas
...
+
+    verifyNumberOfScheduledNotices(1);
+
+    checkOutFixture.checkOutByBarcode(item, requester);
+
+    waitAtMost(1, SECONDS)
+      .until(() -> requestsClient.get(request.getId()).getJson().getString("status"),
+        equalTo(CLOSED_FILLED));
+
+
```
Comment (sherzodn) [inline src/test/java/api/requests/RequestScheduledNoticesProcessingTests.java:185]: extra row can be removed

File: src/test/java/api/requests/RequestScheduledNoticesProcessingTests.java [hunk 2]
```diff
@@ -155,6 +156,41 @@ void uponAtRequestExpirationNoticeShouldNotBeSentWhenRequestExpirationDateHasPas
...
+    IndividualResource request = requestsFixture.place(new RequestBuilder().page()
+      .forItem(item)
+      .withRequesterId(requester.getId())
+      .withRequestDate(getZonedDateTime())
+      .withStatus(OPEN_NOT_YET_FILLED)
+      .withPickupServicePoint(pickupServicePoint)
+      .withRequestExpiration(requestExpiration));
+
+
```
Comment (sherzodn) [inline src/test/java/api/requests/RequestScheduledNoticesProcessingTests.java:176]: extra row can be removed

File: src/main/java/org/folio/circulation/domain/notice/schedule/RequestScheduledNoticeService.java [hunk 3]
```diff
@@ -80,7 +80,7 @@ public Result<CheckInContext> rescheduleRequestNotices(CheckInContext context) {
     return succeeded(context);
   }
 
-  private void rescheduleRequestNotices(Request request) {
+  public void rescheduleRequestNotices(Request request) {
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/domain/notice/schedule/RequestScheduledNoticeService.java:83]: Can we use `rescheduleRequestNotices(RequestAndRelatedRecords relatedRecords)` instead and keep this method private? I get it that we didn't have `relatedRecords` in the `UpdateRequestQueue`, but we will in the resource.

File: src/main/java/org/folio/circulation/domain/UpdateRequestQueue.java [hunk 4]
```diff
@@ -50,7 +54,7 @@ public static UpdateRequestQueue using(Clients clients,
 
     return new UpdateRequestQueue(requestQueueRepository,
       requestRepository, new ServicePointRepository(clients),
-      new ConfigurationRepository(clients));
+      new ConfigurationRepository(clients), RequestScheduledNoticeService.using(clients));
```
[the remark this answers is not in the document]
Comment (humoyunepam) [reply src/main/java/org/folio/circulation/domain/UpdateRequestQueue.java:57]: That didn't satisfy me too, but I couldn't find possible way. Well, will try once more.

**Pull request merged**